modin/pandas/dataframe.py

```diff
@@ -2993,9 +2993,8 @@ def _create_or_update_from_compiler(
         DataFrame or None
             None if update was done, ``DataFrame`` otherwise.
         """
-        assert (
-            isinstance(new_query_compiler, type(self._query_compiler))
```

Comment:
Why removed?

Reply:
new_query_compiler can also be an instance of a different query compiler(Nativequerycompiler or PandasQueryCompiler), 

Eg during operations like insert in the  _create_or_update_from_compiler the constructor gets called directly [eg](https://github.com/arunjose696/modin/blob/arun-sqc-interop/modin/pandas/base.py#L562) , 

Thus it would be normal for the cases where constructor may return a new query compiler for a case where user changes the query compiler mode between creating data_frame and insert operation